server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -754,14 +754,9 @@ public void run() {
     mincMetrics = new TabletServerMinCMetrics();
     ceMetrics = new CompactionExecutorsMetrics();
     MetricsUtil.initializeProducers(metrics, updateMetrics, scanMetrics, mincMetrics, ceMetrics);
-
-    this.compactionManager = new CompactionManager(new Iterable<Compactable>() {
-      @Override
-      public Iterator<Compactable> iterator() {
-        return Iterators.transform(onlineTablets.snapshot().values().iterator(),
-            Tablet::asCompactable);
-      }
-    }, getContext(), ceMetrics);
+    var tabletIterator = onlineTablets.snapshot().values().iterator();
+    Iterable<Compactable> compactables = () -> transform(tabletIterator, Tablet::asCompactable);
```

Comment:
This changes the behavior from getting a snapshot each time the iterator is requested to only getting a snapshot once ever.  Also if the iterator over the snapshot is only created once, then once its exhausted its seem like this iterable will always return an empty iterator.  I think the previous behavior needs to be maintained and the following might do it.

```suggestion
    Iterable<Compactable> compactables = () -> transform( onlineTablets.snapshot().values().iterator(), Tablet::asCompactable);
```

Reply:
Good catch. I don't see how yours is any different though.